daq/runner.py

```diff
@@ -55,19 +56,33 @@ class Devices:
     """Container for all devices"""
     def __init__(self):
         self._devices = {}
+        self._set_ids = set()
 
-    def new_device(self, mac):
+    def new_device(self, mac, port_info):
```

Comment:
I think it's cleaner to use def new_device(self, mac, port_info=None):
so you don't have to pass None when it's not needed.

Reply:
agreed -- python isn't my native language so still use old-style constructs sometimes!